src/util.rs

```diff
@@ -183,7 +183,9 @@ pub fn access_of(register: &Register) -> Access {
         if let Some(ref fields) = register.fields {
             if fields.iter().all(|f| f.access == Some(Access::ReadOnly)) {
                 Access::ReadOnly
-            } else if fields.iter().all(|f| f.access == Some(Access::WriteOnly)) {
+            } else if fields.iter().all(|f| f.access == Some(Access::WriteOnce)) {
+                Access::WriteOnce
+            } else if fields.iter().all(|f| f.access == Some(Access::WriteOnly) || f.access == Some(Access::WriteOnce)) {
```

Comment:
I think this second `f.access == Some(Access::WriteOnce)` comparison is redundant

Reply:
This checks that all the fields are either `WriteOnly` or `WriteOnce`, instead of only `WriteOnly`. A register that has both `WriteOnly` and `WriteOnce` fields is still `WriteOnly` and not `ReadWrite`.

Reply:
> A register that has both `WriteOnly` and `WriteOnce` fields is still `WriteOnly` and not `ReadWrite`.

Yes, but any register with both `WriteOnly` and `WriteOnce` will be processed by first `WriteOnce` `if` statement. The next `if` statement will process just `WriteOnly` registers. Or not?